tests/integration/Migration/MigrationFix/SwagMigrationEntityTest.php

```diff
@@ -0,0 +1,130 @@
+
+use PHPUnit\Framework\Attributes\DataProvider;
+use PHPUnit\Framework\TestCase;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepository;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\Test\TestCaseBase\DatabaseTransactionBehaviour;
+use Shopware\Core\Framework\Test\TestCaseBase\KernelTestBehaviour;
+use Shopware\Core\Framework\Uuid\Uuid;
+use SwagMigrationAssistant\Migration\Connection\SwagMigrationConnectionCollection;
+use SwagMigrationAssistant\Migration\Mapping\MappingService;
+use SwagMigrationAssistant\Migration\MigrationContext;
+use SwagMigrationAssistant\Migration\MigrationFix\SwagMigrationFixCollection;
+use SwagMigrationAssistant\Migration\MigrationFix\SwagMigrationFixEntity;
+use SwagMigrationAssistant\Profile\Shopware\Gateway\Local\ShopwareLocalGateway;
+
+class SwagMigrationEntityTest extends TestCase
+{
+    use DatabaseTransactionBehaviour;
+    use KernelTestBehaviour;
+
+    /**
+     * @var EntityRepository<SwagMigrationFixCollection>
+     */
+    private EntityRepository $migrationFixRepository;
+
+    /**
+     * @var EntityRepository<SwagMigrationConnectionCollection>
+     */
+    private EntityRepository $connectionRepository;
+
+    private MappingService $mappingService;
+
+    protected function setUp(): void
+    {
+        $this->migrationFixRepository = $this->getContainer()->get('swag_migration_fix.repository');
+        $this->connectionRepository = $this->getContainer()->get('swag_migration_connection.repository');
+        $this->mappingService = $this->getContainer()->get(MappingService::class);
+    }
+
+    #[DataProvider('valueData')]
+    public function testGetAndSetValue(mixed $value): void
+    {
+        $context = Context::createDefaultContext();
+        $connectionId = $this->createConnection($context);
+        $mappingId = $this->createMapping($connectionId);
+
+        $fixId = Uuid::randomHex();
+
+        $migrationFix = new SwagMigrationFixEntity();
+        $migrationFix->setId($fixId);
+        $migrationFix->setConnectionId($connectionId);
+        $migrationFix->setMainMappingId($mappingId);
+        $migrationFix->setPath('this.is.any.path');
+        $migrationFix->setValue($value);
+
+        $this->migrationFixRepository->upsert([$migrationFix->jsonSerialize()], $context);
+
+        $criteria = new Criteria([$fixId]);
+
+        $result = $this->migrationFixRepository->search($criteria, $context)->getEntities()->first();
+
+        static::assertNotNull($result);
+        static::assertSame($value, $result->getValue());
+    }
+
+    /**
+     * @return array<string, array<string, mixed>>
+     */
+    public static function valueData(): array
+    {
+        return [
+            'integer' => ['value' => 12],
+            'float' => ['value' => 42.12],
+            'string' => ['value' => 'fooBar'],
+            'array' => ['value' => ['key' => 'value']],
+        ];
+    }
```

Comment:
Isn't the value that is returned here and passed as an argument to the above method always an array like `['value' => 12]` instead of just a number `12`?

Reply:
I think this makes it much more complicated. Also, we use plain SQL to get the fix.

Reply:
You always have to get the value out of the object. Thats unnecessary and complicate. If you dont know that, you have to investigate further.

I like the idea with the DAL Field extension. This should do the work for us